java/vector/src/test/java/org/apache/arrow/vector/file/json/TestJSONFile.java

```diff
@@ -156,6 +161,48 @@ public void testWriteReadDateTimeJSON() throws IOException {
   }
 
   @Test
+  public void testWriteReadDictionaryJSON() throws IOException {
```

Comment:
Can you expand this test to include

* More than 1 dictionary
* Two vectors using the same dictionary